api/schema.py

```diff
@@ -0,0 +1,178 @@
+                .where(ORM.Marker.lat.between(lo = minLat, hi = maxLat))
+                .where(ORM.Marker.lng.between(lo = minLng, hi = maxLng))
+                .limit(perPage)
+                .offset(offset)
+                .execute()
+        )
+
+    @strawberry.field
+    async def propertySearchInRange(
+        pos: GeoPoint,
+        range: float = 1,
+        rangeUnit: RangeUnit = "km",
+        perPage: int = 100,
+        page: int = 1,
+    ) -> typing.List[Property]:
+        return []
+
+    @strawberry.field
+    async def incidentSearchInRange(
+        pos: GeoPoint,
+        range: float = 1,
+        rangeUnit: RangeUnit = "km",
+        perPage: int = 100,
+        page: int = 1,
+    ) -> typing.List[Incident]:
+        offset = (page - 1) * perPage
+        range /= 100
+
+        minLat = pos.lat - range
+        maxLat = pos.lat + range
+        minLng = pos.lng - range
+        maxLng = pos.lng + range
+
+        return (
+            ORM.Incident.select()
+                .where(ORM.Incident.lat.between(lo = minLat, hi = maxLat))
+                .where(ORM.Incident.lng.between(lo = minLng, hi = maxLng))
+                .limit(perPage)
+                .offset(offset)
+                .execute()
+        )
+
+    @strawberry.field
+    async def postcodeSearch(
+        pattern: str,
+        perPage: int = 100,
+        page: int = 1,
+    ) -> typing.List[Postcode]:
+        return []
+
+    @strawberry.field
+    async def propertySearch(
+        postcodePattern: str,
+        dateFrom: typing.Optional[str] = None,
+        dateTo: typing.Optional[str] = None,
+        perPage: int = 100,
+        page: int = 1
+    ) -> typing.List[Property]:
+        return []
+
+    @strawberry.field
+    async def transactionSearch(
+        postcodePattern: str,
+        dateFrom: typing.Optional[str] = None,
+        dateTo: typing.Optional[str] = None,
+        perPage: int = 100,
+        page: int = 1
+    ) -> typing.List[Transaction]:
+        return []
+
+    @strawberry.field
+    async def timelineSearch(
+        postcodePattern: str,
+        postcodes: typing.List[str],
+        dateFrom: typing.Optional[str] = None,
+        dateTo: typing.Optional[str] = None,
+        perPage: int = 100,
+        page: int = 1
+    ) -> typing.List[Timeline]:
+        offset = (page - 1) * perPage
+        query = (
+            ORM.Timeline.select()
+                .limit(perPage)
+                .offset(offset)
+        )
+
+        if postcodePattern:
+            query = (
+                query
+                    .orwhere(ORM.Timeline.postcode.contains(postcodePattern))
+            )
+
+        if postcodes:
+            query = (
+                query
+                    .orwhere(ORM.Timeline.postcode.in_(postcodes))
+            )
+
+        if dateFrom and dateTo:
+            query = (
+                query
+                    .orwhere(ORM.Timeline.date.between(lo = dateFrom, hi = dateTo))
```

Comment:
I am not expert here, but try 

```
        if dateFrom :
            query = (
                query
                    .where(ORM.Timeline.date.__ge__(dateFrom))
            )

        if dateTo:
            query = (
                query
                    .where(ORM.Timeline.date.__le__(dateTo))
            )
```          

i think this should cover between by itself :)